apps/studio/src/components/Navigation.tsx

```diff
@@ -67,6 +68,142 @@ const ServersNavigation: React.FunctionComponent<NavigationSectionProps> = ({
   );
 };
 
+const ChannelsNavigation: React.FunctionComponent<NavigationSectionProps> = ({
+  document,
+  hash,
+}) => {
+  const { navigationSvc } = useServices();
+
+  const channels = document.channels().all().map(
+    (channel) => {
+      return  <li
+        key={channel.id()}
+        className={`p-2 pl-3 text-white cursor-pointer text-xs border-t border-gray-700 hover:bg-gray-900 ${
+          hash === `channels-${channel.id()}` ? 'bg-gray-800' : ''
+        }`}
+        onClick={() =>
+          navigationSvc.scrollTo(
+            `/channels/${(channel.id() ?? '').replace(/\//g, '~1')}`,
+            `channels-${channel.id()}`,
+          )
+        }
+      >
+        <div className="flex flex-row">
+          <div className="flex-none">
+            <span className="mr-3 text-xs uppercase text-blue-500 font-bold">
+              {channel.id()}
+            </span>
+          </div>
+          <span className="truncate">{channel.address()}</span>
+        </div>
+      </li>
+    },
+  );
+
+  return (
+    <>
+      <div
+        className={`p-2 pl-3 text-white cursor-pointer hover:bg-gray-900 ${
+          hash === 'channels' ? 'bg-gray-800' : ''
+        }`}
+        onClick={() =>
+          navigationSvc.scrollTo(
+            '/channels',
+            'channels',
+          )
+        }
+      >
+        Channels
+      </div>
+      <ul>{channels}</ul>
+    </>
+  );
+};
+
+const V3OperationsNavigation: React.FunctionComponent<NavigationSectionProps> = ({
```

Comment:
@KhudaDad414 we will have more and more duplicated code as we are in the transition period to V3. 
We should just add an exception in SonarCloud, and clean after ending support for V2.
Not sure if there is an other clean way to do it.